tests/bdd/features/rebuild/test_log_based_rebuild.py

```diff
@@ -0,0 +1,297 @@
+POOL_2_UUID = "91a60318-bcfe-4e36-92cb-ddc7abf212ea"
+POOL_3_UUID = "4d471e62-ca17-44d1-a6d3-8820f6156c1a"
+POOL_4_UUID = "d5c5e3de-d77b-11ed-afa1-0242ac120002"
+RECONCILE_PERIOD_SECS = 1
+FIO_RUNTIME = 120
+faulted_child_uri = None
+
+
+@pytest.fixture(autouse=True)
+def init():
+    Deployer.start(
+        io_engines="4",
+        wait="10s",
+        reconcile_period=f"{RECONCILE_PERIOD_SECS}s",
+        # faulted_child_wait_period="30s",
+        cache_period="1s",
+    )
+
+    # Only create 3 pools so we can control where the intial replicas are placed.
+    ApiClient.pools_api().put_node_pool(
+        NODE_1_NAME, POOL_1_UUID, CreatePoolBody(["malloc:///disk?size_mb=50"])
+    )
+    ApiClient.pools_api().put_node_pool(
+        NODE_2_NAME, POOL_2_UUID, CreatePoolBody(["malloc:///disk?size_mb=50"])
+    )
+    ApiClient.pools_api().put_node_pool(
+        NODE_3_NAME, POOL_3_UUID, CreatePoolBody(["malloc:///disk?size_mb=50"])
+    )
+
+    yield
+    Deployer.stop()
+
+
+@scenario(
+    "log-based-rebuild.feature",
+    "Faulted child is not online again within timed-wait period",
+)
+def test_faulted_child_is_not_online_again_within_timedwait_period():
+    """Faulted child is not online again within timed-wait period."""
+
+
+@scenario(
+    "log-based-rebuild.feature",
+    "Faulted child is online again within timed-wait period",
+)
+def test_faulted_child_is_online_again_within_timedwait_period():
+    """Faulted child is online again within timed-wait period."""
+
+
+@scenario(
+    "log-based-rebuild.feature",
+    "Node goes permanently down while log-based rebuild running",
+)
+def test_node_goes_permanently_down_while_logbased_rebuild_running():
+    """Node goes permanently down while log-based rebuild running."""
+
+
+@scenario(
+    "log-based-rebuild.feature",
+    "Volume target is moved before starting log-based rebuild",
+)
+def test_volume_target_is_moved_before_starting_logbased_rebuild():
+    """Volume target is moved before starting log-based rebuild."""
+
+
+@scenario(
+    "log-based-rebuild.feature", "Volume target is moved during log-based rebuild"
+)
+def test_volume_target_is_moved_during_logbased_rebuild():
+    """Volume target is moved during log-based rebuild."""
+
+
+@given("a volume with three replicas, filled with user data")
+def _():
+    """a volume with three replicas, filled with user data."""
+    request = CreateVolumeBody(
+        VolumePolicy(True), NUM_VOLUME_REPLICAS, VOLUME_SIZE, False
+    )
+    ApiClient.volumes_api().put_volume(VOLUME_UUID, request)
+    volume = ApiClient.volumes_api().put_volume_target(
+        VOLUME_UUID,
+        publish_volume_body=PublishVolumeBody(
+            {}, Protocol("nvmf"), node=NODE_1_NAME, frontend_node=""
+        ),
+    )
+
+    # Now the volume has been created, create an additional pool.
+    ApiClient.pools_api().put_node_pool(
+        NODE_4_NAME, POOL_4_UUID, CreatePoolBody(["malloc:///disk?size_mb=50"])
```

Comment:
Will need tempfs disks for persistence

Reply:
modified to use tmpfs disks exported via aio//